src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter(animal => animal.health > 0);
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herb) {
+    if (herb.hidden === false) {
```

Comment:
```suggestion
    if (!herb.hidden) {
```

Reply:
if I try change condition but i got error / 'bite' method shouldn't decrease 'health' property of the carnivores/ I think one of the tests compares the value true and false but none of the others, for example undefined, so in the condition I use false.

Reply:
you just need to add validation that your herb its Herbivore but not Carnivore